h/api/search/core.py

```diff
@@ -18,28 +23,62 @@ def search(request, params, private=True):
         results
     :type private: bool
 
-    :returns: a dict with keys "rows" (the list of matching annotations, as
-        dicts) and "total" (the number of matching annotations, an int)
+    :param separate_replies: Whether or not to include a "replies" key in the
+        result containing a list of all replies to the annotations in the
+        "rows" key. If this is True then the "rows" key will include only
+        top-level annotations, not replies.
+    :type private: bool
+
+    :returns: A dict with keys:
+      "rows" (the list of matching annotations, as dicts)
+      "total" (the number of matching annotations, an int)
+      "replies": (the list of all replies to the annotations in "rows", if
+        separate_replies=True was passed)
     :rtype: dict
     """
-    builder = query.Builder()
-
-    builder.append_filter(query.AuthFilter(request, private=private))
-    builder.append_filter(query.UriFilter())
-    builder.append_filter(lambda _: \
-        nipsa.nipsa_filter(request.authenticated_userid))
-    builder.append_filter(query.GroupFilter())
+    def make_builder():
+        builder = query.Builder()
+        builder.append_filter(query.AuthFilter(request, private=private))
+        builder.append_filter(query.UriFilter())
+        builder.append_filter(
+            lambda _: nipsa.nipsa_filter(request.authenticated_userid))
+        builder.append_filter(query.GroupFilter())
+        builder.append_matcher(query.AnyMatcher())
+        builder.append_matcher(query.TagsMatcher())
+        return builder
 
-    builder.append_matcher(query.AnyMatcher())
-    builder.append_matcher(query.TagsMatcher())
-
-    body = builder.build(params)
-    results = models.Annotation.search_raw(body,
+    builder = make_builder()
+    if separate_replies:
+        builder.append_filter(query.TopLevelAnnotationsFilter())
+    results = models.Annotation.search_raw(builder.build(params),
                                            raw_result=True,
                                            authorization_enabled=False)
 
     total = results['hits']['total']
     docs = results['hits']['hits']
     rows = [models.Annotation(d['_source'], id=d['_id']) for d in docs]
+    return_value = {"rows": rows, "total": total}
```

Comment:
Minor niggle, but `return_value` isn't a particularly helpful name. I can see it's a return value because it follows the `return` keyword, but that doesn't tell me what it is.